projects/vdk-heartbeat/src/vdk/internal/heartbeat/job_controller.py

```diff
@@ -379,14 +379,27 @@ def check_job_execution_finished(self) -> Optional[str]:
         )
         if not execution_list:
             return None
-        latest_end_date = max(
-            datetime.fromisoformat(e["end_time"]) for e in execution_list
-        )
-        return [
-            e["status"]
-            for e in execution_list
-            if datetime.fromisoformat(e["end_time"]) == latest_end_date
-        ].pop()
+
+        try:
+            latest_end_date = max(
+                datetime.fromisoformat(str(e["end_time"])) for e in execution_list
+            )
+            return [
+                e["status"]
+                for e in execution_list
+                if datetime.fromisoformat(str(e["end_time"])) == latest_end_date
+            ].pop()
+        except ValueError as e:
+            log.info(
+                "Something happened while checking the end_time of the "
```

Comment:
'Something happened' is very vague. Let's say what happened. 

Also ValueError could be thrown from other places potentially . if you want to make sure datetime is parsed without exception have a function `datatime_from_iso_no_except` 

Generally a good principle for exceptions is to catch an exception as close as possible to its source

Reply:
Done, thanks